src/main/java/net/openhft/chronicle/wire/Wires.java

```diff
@@ -476,18 +478,16 @@ public static int removeMaskedTidFromHeader(final int header) {
     public static <E> E objectMap(ValueIn in, @Nullable E using, @Nullable Class clazz, SerializationStrategy<E> strategy) {
         if (in.isNull())
             return null;
-        boolean nullObject = false;
         if (clazz == Object.class)
             strategy = MAP;
         if (using == null) {
             using = (E) strategy.newInstanceOrNull(clazz);
-            nullObject = using == null;
         }
         if (Throwable.class.isAssignableFrom(clazz))
             return (E) WireInternal.throwable(in, false, (Throwable) using);
 
-        if (using == null && !nullObject)
-            throw new IllegalStateException("failed to create instance of clazz=" + clazz + " is it aliased?");
+        if (using == null)
+            throw new FatalMarshallingException("failed to create instance of clazz=" + clazz + " is it aliased?");
```

Comment:
This ended up being effectively unreachable from what I could see (it was effectively `if (using == null && using != null)` if you look at what happened up till this point. It meant we ended up passing in `null` to some @NotNull parameters below.

It became that way here https://github.com/OpenHFT/Chronicle-Wire/commit/4d69ee9e7d5cdb40774307d071fb53a568bbf37d#diff-93e71c0374096bba5af3369c6b8ec64a71bad073b960cbefe8ec30703cf81577R436

Reply:
Another odd thing is a few lines down it calls `ValueIn#marshallable(Object, SerializationStrategy)`... the `BinaryValueIn` implementation of that has the first parameter marked as `@Nullable` but all other implementations have it marked as `@NotNull`. I'm going with the majority here and assuming we don't want to pass this point if `using == null`